src/kvm_bindings_versionize/src/lib.rs

```diff
@@ -8,10 +8,77 @@
 #[macro_use]
 extern crate vmm_sys_util;
 
+extern crate kvm_bindings;
 extern crate versionize;
 extern crate versionize_derive;
 
 #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
 mod x86;
+
 #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
 pub use self::x86::*;
+
+use std::mem::{align_of, size_of};
+use std::ptr;
+
+use vmm_sys_util::fam::{FamStruct, FamStructWrapper};
+
+fn convert_bitwise<T, U>(t: &T) -> U {
+    assert_eq!(align_of::<T>(), align_of::<U>());
+    assert_eq!(size_of::<T>(), size_of::<U>());
+    // Safe because `src` is aligned, the two types have the same size and alignment,
+    // and they are both plain old data structs.
```

Comment:
Is there a way to guarantee this? Maybe force them through a dummy FFI or something similar to make sure we don't accidentally break this assumption?

My worry is that the function safety is based on only using it for POD or `repr(C)` data, but that isn't in any way enforced, and we could accidentally misuse it.

Reply:
The assumption must hold for kvm bindings, and the unit tests and sanity checks above verify everything except the presence of `#[repr(C)]`, which is where the dummy FFI might come in handy for extra certainty.